lib/web/apiserver_test.go

```diff
@@ -2566,6 +2566,81 @@ func TestCheckAccessToRegisteredResource(t *testing.T) {
 	}
 }
 
+func TestAuthExport(t *testing.T) {
+	env := newWebPack(t, 1)
+	clusterName := env.server.ClusterName()
+
+	proxy := env.proxies[0]
+	pack := proxy.authPack(t, "test-user@example.com", nil)
+
+	for _, tt := range []struct {
+		name           string
+		authType       string
+		expectedStatus int
+	}{
+		{
+			name:           "all",
+			authType:       "",
+			expectedStatus: http.StatusOK,
+		},
+		{
+			name:           "host",
+			authType:       "host",
+			expectedStatus: http.StatusOK,
+		},
+		{
+			name:           "user",
+			authType:       "user",
+			expectedStatus: http.StatusOK,
+		},
+		{
+			name:           "windows",
+			authType:       "windows",
+			expectedStatus: http.StatusOK,
+		},
+		{
+			name:           "db",
+			authType:       "db",
+			expectedStatus: http.StatusOK,
+		},
+		{
+			name:           "invalid",
+			authType:       "invalid",
+			expectedStatus: http.StatusBadRequest,
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			// export host certificate
+			endpointExport := pack.clt.Endpoint("webapi", "sites", clusterName, "auth", "export")
+
+			if tt.authType != "" {
+				endpointExport = fmt.Sprintf("%s?type=%s", endpointExport, tt.authType)
+			}
+
+			req, err := http.NewRequest(http.MethodGet, endpointExport, nil)
```

Comment:
Please use the context-aware variants (along with a timeout-configured context).